In [52]:
import pandas as pd

data = pd.read_csv( 'Nationale ORCiD_NL Monitor.csv' )

## Unique organisations
organisations = data['Selecteer je Universiteit'].unique()

def sortedByValue( dict ):      
    return sorted( dict , key=lambda x: dict[x]) 



# Ambitionist

De 'Ambitionist' is de prijs voor het hoogste aantal geregistreerde ORCIDS in het CRIS. Alle universiteiten hebben vooraf een bepaald ambitieniveau gedefinieerd, en de 'ambitionist' is de universiteit die deze ambitie het beste heeft weten te realiseren. 

De scores in deze categorie zijn berekend door de kijken naar de cijfers van het meest recente meetmoment, en door het aantal geregistreerde ORCID IDs te deen door het totaal aantal actieve onderzoekers. Dit cijfer geeft een indicatie van de total ORCID-uptake. Hoeveel procent van de onderzoekers heeft een ORCID ingevuld in het CRIS? 

In [53]:
ambitionist_scores = dict()

for o in organisations:
    data_inst = data[ data['Selecteer je Universiteit'] == o ]   
    
    # Meest recente datum
    dateMax = data_inst['Datum van meting'].max()
    mostRecent = data_inst[ data_inst['Datum van meting'] == dateMax ].squeeze()
    score = mostRecent['Aantal ORCiD registraties in het CRIS van Onderzoekers'] / mostRecent['Aantal onderzoekers']
    ambitionist_scores[o] = score
    print( f'{o}\nStand van zaken op {dateMax}:') 
    print( f"Aantal ORCID registraties: {mostRecent['Aantal ORCiD registraties in het CRIS van Onderzoekers']}\nAantal actieve onderzoekers: {mostRecent['Aantal onderzoekers']}" )
    print( f'Hun score komt daarmee op {score}.\n')

VU - Vrije Universiteit Amsterdam
Stand van zaken op 2020-10-15:
Aantal ORCID registraties: 1384
Aantal actieve onderzoekers: 2936
Hun score komt daarmee op 0.4713896457765668.

TU - Tilburg University
Stand van zaken op 2019-12-11:
Aantal ORCID registraties: 325
Aantal actieve onderzoekers: 2102
Hun score komt daarmee op 0.15461465271170313.

TUD - Delft University of Technology
Stand van zaken op 2020-10-06:
Aantal ORCID registraties: 2802
Aantal actieve onderzoekers: 6646
Hun score komt daarmee op 0.42160698164309357.

WUR - Wageningen University & Research
Stand van zaken op 2020-10-13:
Aantal ORCID registraties: 1033
Aantal actieve onderzoekers: 2004
Hun score komt daarmee op 0.5154690618762475.

UM - Universiteit Maastricht
Stand van zaken op 2020-10-14:
Aantal ORCID registraties: 1228
Aantal actieve onderzoekers: 5230
Hun score komt daarmee op 0.23479923518164436.

UT - University of Twente
Stand van zaken op 2020-10-12:
Aantal ORCID registraties: 595
Aantal actieve onderzoekers

De hoogste score is behaald door de **Universiteit Leiden**. Van de 2654 onderzoekers hebben uiteindelijk 2273 een ORCID ingevoerd in het CRIS. Het uptake-percentage is daarmee 85.6 %.

De volledige lijst van de behaalde scores is hieronder te bekijken.

In [54]:
for ub in reversed( sortedByValue( ambitionist_scores) ):
    print( f'{ub}: { ambitionist_scores[ub]  } ' )
    
    
print('\n\n')    
    

UL - Universiteit Leiden: 0.8564431047475509 
WUR - Wageningen University & Research: 0.5154690618762475 
VU - Vrije Universiteit Amsterdam: 0.4713896457765668 
RUG - Rijksuniversiteit Groningen: 0.45141903171953257 
TUD - Delft University of Technology: 0.42160698164309357 
TUE - Technische Universiteit Eindhoven: 0.3907208588957055 
TiU - Tilburg University: 0.2857817666509211 
UM - Universiteit Maastricht: 0.23479923518164436 
OU - Open Universiteit: 0.22276029055690072 
UT - University of Twente: 0.183585313174946 
TU - Tilburg University: 0.15461465271170313 
KNAW - Koninklijke Nederlandse Academie der Wetenschappen: 0.1352112676056338 
UvA - Universiteit van Amsterdam: 0.06433990895295903 
UMCU - Universitair Medisch Centrum Utrecht: 0.054280779669380705 
UU - Universiteit Utrecht: 0.050029429075927015 





# Connector


De 'Connector' is de prijs voor het grootste percentage koppelingen tussen het CRIS van een universiteit en de ORCID database. Wanneer dit soort koppelingen tot stand zijn gebracht kunnen er gegevens worden geëxporteerd vanuit het CRIS naar de ORCID-records van onderzoekers, en in sommige gevallen ook vice versa. Onderzoekers moeten deze koppelingen zelf activeren. Bibliotheken kunnen dit niet namens onderzoekers doen. Wanneer er in ORCID-campagnes ook acties zijn uitgevoerd om onderzoekers te bewegen om deze koppeingen tot stand te brengen, naast alleen het registreren van de ORCIDs, dan verdient dit een extra beloning. Met een actieve ORCiD-koppeling kunnen een onderzoeker uiteindelijk meer. Via dit soort koppelingen kan uiteindelijk de meerwaarde van ORCID worden aangetoond. 

De scores voor deze prijs zijn berekend door het aantal koppelingen in de meest recente meting te delen door het aantal onderzoekers op dat moment. 



In [55]:
connector_scores = dict()


for o in organisations:
    data_inst = data[ data['Selecteer je Universiteit'] == o ]  
    
    ## Meest recente meting
    dateMax = data_inst['Datum van meting'].max()
    mostRecent = data_inst[ data_inst['Datum van meting'] == dateMax ].squeeze()
    
    score = mostRecent['Aantal ORCiD Export Koppelingen in het CRIS van Onderzoekers'] / mostRecent['Aantal onderzoekers']
    
    ## Incorrecte resultaten (bijv delingen door 0) worden genegeerd
    if( not( pd.isnull(score) ) ):
        connector_scores[o] = float(score)

        print( f'{o}\nStand van zaken op {dateMax}:') 
        print( f"Aantal ORCiD Export Koppelingen in CRIS: { int(mostRecent['Aantal ORCiD Export Koppelingen in het CRIS van Onderzoekers']) }\nAantal onderzoekers: {mostRecent['Aantal onderzoekers']}" )
        print( f'Hun score komt daarmee op {score}.\n')
    
        

VU - Vrije Universiteit Amsterdam
Stand van zaken op 2020-10-15:
Aantal ORCiD Export Koppelingen in CRIS: 1206
Aantal onderzoekers: 2936
Hun score komt daarmee op 0.41076294277929154.

TUD - Delft University of Technology
Stand van zaken op 2020-10-06:
Aantal ORCiD Export Koppelingen in CRIS: 636
Aantal onderzoekers: 6646
Hun score komt daarmee op 0.09569665964489919.

WUR - Wageningen University & Research
Stand van zaken op 2020-10-13:
Aantal ORCiD Export Koppelingen in CRIS: 115
Aantal onderzoekers: 2004
Hun score komt daarmee op 0.05738522954091816.

UM - Universiteit Maastricht
Stand van zaken op 2020-10-14:
Aantal ORCiD Export Koppelingen in CRIS: 343
Aantal onderzoekers: 5230
Hun score komt daarmee op 0.06558317399617591.

UT - University of Twente
Stand van zaken op 2020-10-12:
Aantal ORCiD Export Koppelingen in CRIS: 147
Aantal onderzoekers: 3241
Hun score komt daarmee op 0.04535637149028078.

UvA - Universiteit van Amsterdam
Stand van zaken op 2020-10-15:
Aantal ORCiD Export 

Wanneer we de universiteiten rangschikken aan de hand van deze scores staat de UL wederom bovenaan. Omdat universiteiten maar een prijs in één categorie kunnen winnen, gaat de trofee in dit geval naar nummer twee van de lijst, en dat is de **Vrije Universiteit Amsterdam**. Op het laatste peilmoment, 15 oktober 2020, hadden 1206 van de 2036 onderoekers een koppeling tt stand gebracht met de ORCID-database. Dit komt neer op 41% van alle actieve onderzoekers. 

In [56]:
def sortedByValue( dict ):      
    return sorted( dict , key=lambda x: dict[x]) 

for ub in reversed( sortedByValue(connector_scores) ):
    print( f'{ub}: { connector_scores[ub]  } ' )
    
   
    
## Berekening rode lantaarn
score = 0
for ub in sortedByValue(connector_scores) :
    score += 1
    if score <= 3:
        red_lantern[ub] = score


UL - Universiteit Leiden: 0.5550113036925396 
VU - Vrije Universiteit Amsterdam: 0.41076294277929154 
TiU - Tilburg University: 0.21020311761927254 
RUG - Rijksuniversiteit Groningen: 0.15592654424040067 
TUD - Delft University of Technology: 0.09569665964489919 
TUE - Technische Universiteit Eindhoven: 0.09164110429447853 
UM - Universiteit Maastricht: 0.06558317399617591 
WUR - Wageningen University & Research: 0.05738522954091816 
UvA - Universiteit van Amsterdam: 0.048558421851289835 
OU - Open Universiteit: 0.048426150121065374 
UT - University of Twente: 0.04535637149028078 
UU - Universiteit Utrecht: 0.028693349028840496 
KNAW - Koninklijke Nederlandse Academie der Wetenschappen: 0.020657276995305163 
UMCU - Universitair Medisch Centrum Utrecht: 0.0 


## Bloomer

De 'bloomer' beloont de universiteit waar de sterkste groei is geweest vanaf de vertrekpositie. Welke uptake percentage had de universiteit aan het begin van de ORCID-campage, en welk percentage is er *uiteindelijk* bereikt? Welke universiteit heeft de meeste progressie laten zien?

De scores in deze categorie zijn berekend door, kijkend naar alle metingen, het laagste aantal ORCIDs en het hoogste aantal ORCIDs te bepalen. Hierbij is er vanuit gegaan dat er alleen maar een groei kan zijn, en dat het laagste aantal ORCIDs inderdaad ook de uitgangspositie is. Beide aantallen zijn gedeeld door het aantal onderoekers, zoals geregistreerd op die tijdstippen. De uiteindelijke score is het verschil tussen deze twee uptake-percentages.  

In [57]:
bloomer_scores = dict()

for o in organisations:
    
    ## Uitgangspositie van 0 wordt verwijderd
    data_inst = data[ data['Selecteer je Universiteit'] == o ] 
    
    data_inst = data_inst[ data_inst['Aantal ORCiD registraties in het CRIS van Onderzoekers'] > 0 ] 
    sorted_df = data_inst.sort_values(by=['Aantal ORCiD registraties in het CRIS van Onderzoekers'], ascending=False)
    
    highest = sorted_df.iloc[0]
    lowest = sorted_df.iloc[-1]
  
    lowest_perc = lowest['Aantal ORCiD registraties in het CRIS van Onderzoekers'] / lowest['Aantal onderzoekers']
    highest_perc = highest['Aantal ORCiD registraties in het CRIS van Onderzoekers'] / highest['Aantal onderzoekers']
    
    bloomer_scores[o] = ( highest_perc - lowest_perc ) * 100
    
    print( f'{o}' )
    print( f'Uptake-percentage bij begin campagne: {lowest_perc}' )
    print( f'Uptake-percentage bij eind campagne: {highest_perc}' )
    print( f'Dit is een groei van { bloomer_scores[o] } procent.\n' )
    
    

    
    

VU - Vrije Universiteit Amsterdam
Uptake-percentage bij begin campagne: 0.20725009154155988
Uptake-percentage bij eind campagne: 0.4713896457765668
Dit is een groei van 26.41395542350069 procent.

TU - Tilburg University
Uptake-percentage bij begin campagne: 0.15461465271170313
Uptake-percentage bij eind campagne: 0.25
Dit is een groei van 9.538534728829687 procent.

TUD - Delft University of Technology
Uptake-percentage bij begin campagne: 0.06277695716395865
Uptake-percentage bij eind campagne: 0.43913109066224165
Dit is een groei van 37.6354133498283 procent.

WUR - Wageningen University & Research
Uptake-percentage bij begin campagne: 0.04723991507430998
Uptake-percentage bij eind campagne: 0.5154690618762475
Dit is een groei van 46.822914680193755 procent.

UM - Universiteit Maastricht
Uptake-percentage bij begin campagne: 0.04534448602245213
Uptake-percentage bij eind campagne: 0.2364229514488582
Dit is een groei van 19.107846542640605 procent.

UT - University of Twente
Uptake-p

Wanneer de score worden gerangschikt wordt duidelijk dat **Wageningen University & Research** de grootse groei heeft doorgemaakt. Het uptake-percentage was 4.7% bij de start van de campagne. In oktober 2020 was er een uptake-percentage van 51.5%. Dit komt neer op een groei van 46.8%. 

In [58]:
for ub in reversed( sortedByValue(bloomer_scores) ):
    print( f'{ub}: { round( bloomer_scores[ub] , 3 ) }' )
    
    
## Berekening rode lantaarn
score = 0
for ub in sortedByValue(bloomer_scores) :
    score += 1
    if score <= 3:
        red_lantern[ub] = score

WUR - Wageningen University & Research: 46.823
TUD - Delft University of Technology: 37.635
OU - Open Universiteit: 32.044
VU - Vrije Universiteit Amsterdam: 26.414
UM - Universiteit Maastricht: 19.108
TiU - Tilburg University: 17.241
TU - Tilburg University: 9.539
RUG - Rijksuniversiteit Groningen: 7.511
TUE - Technische Universiteit Eindhoven: 5.065
KNAW - Koninklijke Nederlandse Academie der Wetenschappen: 3.802
UvA - Universiteit van Amsterdam: 3.384
UL - Universiteit Leiden: 3.299
UT - University of Twente: 2.637
UU - Universiteit Utrecht: 1.826
UMCU - Universitair Medisch Centrum Utrecht: -1.817


# Pioneer

De toename in het aantal ORCIDs kan worden gevolgd door te kijken naa het aantal ORCIDs dat is ingevuld in het CRIS te delen door het totaal aantal actieve onderzoekers. Dit laatste getal kan bijvoorbeeld worden vastgesteld op basis van de gegevens van de P&O afdeling van de universiteit. Dit cijfer zegt niet noodzakelijk iets over het daadwerkelijke aantal ORCIDs. Er zijn namelijk ook onderzoekers die wel al een ORCID hebben, maar deze nog niet hebben geregistreerd in het CRIS. Sommige universiteiten hebben ook geprobeerd om deze groep in kaart te brengen, door direct in de ORCID database te zoeken via de ORCIS 'Name Search' API. De onderzoekers in deze laatste groep konden dan binnen de campagne een meer gericht verzoek krijgen om hun bestaande ORCID te registreren. 

De 'Pioneer' trofee gaat naar de instelling die de meeste onderzoekers heeft opgespoord in de ORCID-database. De scores zijn berekend door het aantal bij ORCID gevonden onderzoekers te delen door het totaal aantal onderzoekers (zoals vastgelegd in het eigen P&O systeem of in LUCRIS).

Niet alle universiteiten hebben deze cijfers verzameld. De universiteiten die geen cijfers hebben aangeleverd over het aantal onderzoekers in de ORCID-database zijn uiteraard buiten beschouwing gelaten. 

De scores staan hieronder:

In [80]:
pioneer_scores = dict()


for o in organisations:
    data_inst = data[ data['Selecteer je Universiteit'] == o ]  
    
    ## Meest recente meting
    dateMax = data_inst['Datum van meting'].max()
    mostRecent = data_inst[ data_inst['Datum van meting'] == dateMax ].squeeze()
    if( not( pd.isnull(mostRecent['Aantal Onderzoekers in de ORCiD database']) ) ) and mostRecent['Aantal Onderzoekers in de ORCiD database'] > 0:
        score = mostRecent['Aantal Onderzoekers in de ORCiD database'] / mostRecent['Aantal onderzoekers'] 
        pioneer_scores[o] = score
        
        print(f'{o}:')
        print(f"Stand van zaken op { mostRecent['Datum van meting']}:")
        print(f"Aantal onderzoekers: { mostRecent['Aantal onderzoekers']}")
        print(f"Aantal ondezoekers in ORCID database: { mostRecent['Aantal Onderzoekers in de ORCiD database']}")
        print(f"Score: { score }\n")

    

TUD - Delft University of Technology:
Stand van zaken op 2020-10-06:
Aantal onderzoekers: 6646
Aantal ondezoekers in ORCID database: 3885.0
Score: 0.5845621426421908

UvA - Universiteit van Amsterdam:
Stand van zaken op 2020-10-15:
Aantal onderzoekers: 6590
Aantal ondezoekers in ORCID database: 3834.0
Score: 0.5817905918057663

UL - Universiteit Leiden:
Stand van zaken op 2020-09-01:
Aantal onderzoekers: 2654
Aantal ondezoekers in ORCID database: 2386.0
Score: 0.8990203466465713

UU - Universiteit Utrecht:
Stand van zaken op 2020-10-14:
Aantal onderzoekers: 6796
Aantal ondezoekers in ORCID database: 3387.0
Score: 0.4983814008240141

OU - Open Universiteit:
Stand van zaken op 2020-12-06:
Aantal onderzoekers: 413
Aantal ondezoekers in ORCID database: 207.0
Score: 0.5012106537530266



De Universiteit Leiden heeft al een prijs ontvangen in de categorie 'Ambitionist'. De trofee gaat daarom naar de tweede universiteit op de lijst, **Delft University of Technology**. 3885 van 6646 Delftse onderzoekers zijn getraceerd in de ORCID-database. Dat is een percentage van 58%. 

In [81]:
for ub in reversed( sortedByValue(pioneer_scores) ):

    print( f'{ub}: { round( pioneer_scores[ub] , 3 ) }' )

UL - Universiteit Leiden: 0.899
TUD - Delft University of Technology: 0.585
UvA - Universiteit van Amsterdam: 0.582
OU - Open Universiteit: 0.501
UU - Universiteit Utrecht: 0.498


## Story TellerHet meest grappige, sappige, ontroerendste ORCiD-wervingsverhaal
a.       Hierbij kijken we naar diegene die het meest grappige, sappige, ontroerendste Orcid-wervingsverhaal heeft.
3.       Hiermee belonen we diegene die niet alleen haar best doet om met onderzoekers in gesprek te gaan, maar ook een persoonlijke band op bouwen en bijzondere relatie leggen met de bibliotheek en haar functie voor de wetenschappelijke gemeenschap.


## Rode lantaarn

Dit is de prijs voor de achterblijver met de minste progressie. Hiermee willen we de achterblijver met de minste progressie aan de schandpaal nagelen, en er voor zorgen dat diegene die hier de komende maanden in de top 5 staat aansporen om een strategie te bepalen en voor een van die andere prijs categorieën te gaan. (Anders formuleren?)

De scores zijn berekend door eerst te kijken welke universiteiten de laagste scores hebben behaald in de categorieën 'Connector', 'Bloomer' en 'Ambitionist'. De universiteiten hebben cijfers ontvangen aan de hand van hun positie op de ranglijsten. Nummer 15 op de lijst kreeg bijvoorbeeld ook het cijfer 15. De universiteiten die slechtste scoorden ontvangen op die manier de hoogste scores in deze specifieke categorie. 

In [74]:
## Berekening rode lantaarn

red_lantern = dict()
number_lowest = 4

def inverseScores( scores ):
    score = len( scores )
    for ub in sortedByValue( scores):
        if score >=  (len( scores ) - number_lowest):
            red_lantern[ub] = red_lantern.get( ub , 0 ) + score
            print( f'{ub}' )
        score = score - 1

print("Laagste scores in de categorie 'Ambitionist': ")
inverseScores( ambitionist_scores ) 

print("\nLaagste scores in de categorie 'Connector': ")
inverseScores( connector_scores ) 

print("\nLaagste scores in de categorie 'Bloomer': ")
inverseScores( bloomer_scores ) 
        


Laagste scores in de categorie 'Ambitionist': 
UU - Universiteit Utrecht
UMCU - Universitair Medisch Centrum Utrecht
UvA - Universiteit van Amsterdam
KNAW - Koninklijke Nederlandse Academie der Wetenschappen
TU - Tilburg University

Laagste scores in de categorie 'Connector': 
UMCU - Universitair Medisch Centrum Utrecht
KNAW - Koninklijke Nederlandse Academie der Wetenschappen
UU - Universiteit Utrecht
UT - University of Twente
OU - Open Universiteit

Laagste scores in de categorie 'Bloomer': 
UMCU - Universitair Medisch Centrum Utrecht
UU - Universiteit Utrecht
UT - University of Twente
UL - Universiteit Leiden
UvA - Universiteit van Amsterdam


**KNAW** of **University of Twente**?

In [75]:
for ub in reversed( sortedByValue(red_lantern) ):
    print( f'{ub}: { red_lantern[ub] }' )

UMCU - Universitair Medisch Centrum Utrecht: 43
UU - Universiteit Utrecht: 41
KNAW - Koninklijke Nederlandse Academie der Wetenschappen: 25
UT - University of Twente: 24
UvA - Universiteit van Amsterdam: 24
UL - Universiteit Leiden: 12
TU - Tilburg University: 11
OU - Open Universiteit: 10
